keytree/src/lib.rs

```diff
@@ -48,6 +52,35 @@ impl Xprv {
     }
 }
 
+impl Xpub {
+    /// Returns a intermediate child pubkey.
+    pub fn to_intermediate_key(&self, customize: impl FnOnce(&mut Transcript)) -> Xpub {
+        let mut t = Transcript::new(b"Keytree.derivation");
+        t.commit_bytes(b"pt", self.precompressed_pubkey.as_bytes());
+        t.commit_bytes(b"dk", &self.dk);
+
+        // the user can pass customize in so that they can commit arbitrary
+        // data (like an account ID)
+        customize(&mut t);
+
+        // squeeze a challenge scalar
+        let f = t.challenge_scalar(b"f.intermediate");
+
+        // squeeze a new derivation key
+        let mut dk = [0u8; 32];
```

Comment:
maybe call it `dk2` or `child_dk` to disambiguate from `self.dk`?
```suggestion
        let mut child_dk = [0u8; 32];
```